test/dates/io.jl

```diff
@@ -77,7 +94,7 @@ c3 = "96:1:01"
 @test Dates.DateTime(c3,f) + Dates.Year(1900) + Dates.Day(14) == dt
 @test Dates.format(dt-Dates.Day(14),"yy:m:dd") == c3
 c4 = "1996:15:01 # random comment"
-@test Dates.DateTime(c4,f) == dt
+@test_throws ArgumentError Dates.DateTime(c4,f)
```

Comment:
I agree that it should be a pre-processing step.

I think we should merge this now and then open a PR against 0.5 branch with changes that make it possible to backport this.